osu.Game.Rulesets.Osu.Tests/TestSceneOsuScoreProcessor.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Testing;
+using osu.Game.Beatmaps;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Game.Rulesets.Osu.Scoring;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Tests.Visual;
+
+namespace osu.Game.Rulesets.Osu.Tests
+{
+    [HeadlessTest]
+    public partial class TestSceneOsuScoreProcessor : OsuTestScene
+    {
+        [Test]
+        public void TestMinimumRankNeverReachesSRank()
```

Comment:
This should read "never reaches S rank before completion". That should have test coverage too.

Reply:
Maybe should make the minimum rank calculations separate into a virtual method, and can be overwritten when needed, (e.g. the situation that misses or others can affect the rank)